kit/controls/src/UIPillButton/PillButtonContent.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { ColorValue, LayoutAnimation, Platform, StyleSheet, UIManager, View } from 'react-native';
+import Animated, { useAnimatedProps, useAnimatedStyle } from 'react-native-reanimated';
+import { ColorVariants, UILabelAnimated, UILabelRoles } from '@tonlabs/uikit.themes';
+import { UIAnimatedImage } from '@tonlabs/uikit.media';
+import { UIConstant } from '../constants';
+import {
+    UIPillButtonVariant,
+    UIPillButtonIconPosition,
+    ContentColors,
+    BackgroundOverlayColors,
+} from './constants';
+import type { UIPillButtonProps } from './types';
+import { usePressableContentColor } from '../Pressable';
+import { UIIndicator } from '../UIIndicator';
+
+if (Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental) {
+    UIManager.setLayoutAnimationEnabledExperimental(true);
+}
+
+export const PillButtonContent = ({
+    icon,
+    iconPosition = UIPillButtonIconPosition.Left,
+    loading,
+    title,
+    variant = UIPillButtonVariant.Neutral,
+}: UIPillButtonProps) => {
+    const backgroundColor = usePressableContentColor(ContentColors[variant].background);
+    const contentColor = usePressableContentColor(ContentColors[variant].content);
+    const backgroundOverlayColor = usePressableContentColor(BackgroundOverlayColors);
+
+    const animatedContainerStyle = useAnimatedStyle(() => {
+        return {
+            backgroundColor: backgroundColor.value,
+        };
+    });
+
+    const animatedLabelProps = useAnimatedProps(() => {
+        return {
+            color: contentColor.value as ColorValue,
+        };
+    });
+
+    const animatedImageProps = useAnimatedProps(() => {
+        return {
+            tintColor: contentColor.value as ColorValue,
+        };
+    });
+
+    const animatedBackgroundOverlayStyle = useAnimatedStyle(() => {
+        return {
+            backgroundColor: backgroundOverlayColor.value,
+        };
+    });
+
+    React.useLayoutEffect(() => {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+    }, [loading]);
```

Comment:
hm, why do we need it? 🤔

Reply:
To animate button shape transition between loading and regular state on mobile platform.

Reply:
Could you please add it as a comment in the code 🙏 

PS. I'm a bit suspicious to that as it might not work well (mixing reanimated and LayoutAnimations from RN), but AFAIK  it worked fine in this case